### PR TITLE
GTIN extraction, Python 3.12 support

### PR DESCRIPTION
**Pull request by wRAR**

### CODE REVIEW

File: zyte_parsers/gtin.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        gtin = extract_text(node)
+    gtin_id = extract_gtin_id(gtin)
+    gtin_class = gtin_classification(gtin_id)
+    if gtin_class:
+        assert isinstance(gtin_id, str)
+        return Gtin(gtin_class, gtin_id)
+    return None
+
+
+def remove_gtin_numeric_prefix(gtin_code: str) -> str:
```
Comment (kmike) [inline zyte_parsers/gtin.py:61]: Can we make this one private, or should it be a part of public API?

Comment (wRAR) [reply zyte_parsers/gtin.py:61]: Looks like this one is fine to make private.

File: zyte_parsers/gtin.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    "gtin12",
+    "gtin13",
+    "gtin14",
+]
+GTIN_PREFIX_REGEX = re.compile("|".join(GTIN_PREFIX), re.IGNORECASE)
+GTIN_CENTER_REGEX = re.compile(r"^\D*|\D*$")
+
+
+def extract_gtin(node: Union[SelectorOrElement, str]) -> Optional[Gtin]:
+    """Extract a GTIN (Global Trade Item Number) from a node that contains its text.
```
Comment (kmike) [inline zyte_parsers/gtin.py:38]: It seems that unlike other extactors, here we also support text as an input, but it's not documented.

Options:

* document it
* add support for text to other extractors when it makes sense, for consistency
* remove text support, for consistency
* something else?

What do you think?

Comment (wRAR) [reply zyte_parsers/gtin.py:38]: We should document it and add support to other extractors as recently asked in https://github.com/zytedata/zyte-common-items/pull/69#pullrequestreview-1790255775

File: docs/intro.rst [hunk 3]
```diff
@@ -4,7 +4,8 @@ Intro
 
 ``zyte-parsers`` provides functions that extract specific data from HTML
 elements. The input element can be an instance of either
-:class:`parsel.selector.Selector` or :class:`lxml.html.HtmlElement`.
+:class:`parsel.selector.Selector` or :class:`lxml.html.HtmlElement`. Some
+functions also support passing a string as input.
```
Comment (kmike) [inline docs/intro.rst:8]: I think we should clarify what thi string is: should it be a string with HTML to extractd data from, or should it be text extracted from a HTML element?

**Review by kmike — APPROVED** [review]

File: docs/intro.rst (hunk 3 above)
Comment (Gallaecio) [reply docs/intro.rst:8]: > extracted from a HTML element

Or elsewhere (e.g. JSON, regexp).

**Review by Gallaecio — APPROVED** [review]

**Pull request merged**